training-client.mdx

```diff
@@ -69,7 +71,56 @@ input={
 }
 ```
 
-### Input Times
+#### SQL queries as input
+`offline_query` also takes [spine_sql_query](/api-docs#ChalkClient.offline_query.spine_sql_query) as an alternative
+input parameter for retrieving feature values from your offline data store. 
+
+Spine SQL queries are recommended in the following scenarios:
+
+1. You want to retrieve data for multiple rows at once. Chalk will compute an efficient query plan for loading multiple
+rows of data at once. Chalk will also reuse data between rows where appropriate.
+2. You want to query from Chalk as your offline data store. You can reduce unnecessary back-and-forth requests by having
+Chalk execute the SQL query and handle the result rows directly.
+3. You want to use multiple [feature namespaces](/docs/features#namespacing) in your input. Spine SQL queries accept
+multiple feature namespaces as column names.
+
+`spine_sql_query` expects each column in the result to correspond to a feature. Chalk replaces `__` in column names with
```

Comment:
Columns are also allowed to be "pass-through", by requesting them as outputs of the Chalk offline query. For example, if you have a column like "blah" that doesn't correspond to any Chalk feature, as long as you include `target_features=[..., "blah", ...]` the feature is passed through to the output.